src/components/SignUpProfile/SignUpProfile.jsx

```diff
@@ -43,19 +50,54 @@ export default function SignUpProfile(props) {
         const userAllergies = allergiesChecked.map(val => 
             val.toLowerCase()
         )
+        const userLoc = {
+            lat: userLocation.coords.latitude,
+            long: userLocation.coords.longitude
+        }
         // update user profile in the database
         const userProfile = {
             primDiet: userPrimDiets,
             diets: userDiets,
-            allergies: userAllergies
+            allergies: userAllergies,
+            location: userLoc
         }
         const docRef = doc(db, "users", currentUser.uid)
-        updateDoc(docRef, userProfile)
-        .catch(error => {
-            setError(error.message)
+        const promises = []
+        promises.push(
+            updateDoc(docRef, userProfile)
+            .catch(error => {
+                setError(error.message)
+            })
+        )
+        Promise.all(promises)
+        .then(() => {
+            navigate("/profile")
         })
-        navigate("/profile")
-        setLoading(false)
+        .catch(() => {
+            setError("failed to update account")
+        }).finally(() => {
+            setLoading(false)
+        })
+    }
+    
+    // get user's current location
+    function getLocation() {
+        if ("geolocation" in navigator) {
+            navigator.geolocation.getCurrentPosition(
+                function(position) {
+                    setUserLocation(position)
+                    setIsLocationLoading(false)
+                },
+                function(error) {
+                    setError(error.message)
+                    setUserLocation({})
+                    setIsLocationLoading(false)
+                }
+            )
+        } else {
+            setError("please enable location access to find closest sellers")
```

Comment:
You may also want to allow the users to provide a zip code or something as an alternative if they don't want to share their actual location.